test/puppetlabs/puppetdb/scf/migrate_test.clj

```diff
@@ -645,3 +646,227 @@
+                               :character_maximum_length nil
+                               :numeric_precision nil
+                               :numeric_precision_radix nil
+                               :data_type "bytea"
+                               :column_name "value_hash"
+                               :table_name "fact_values"}
+                              :right-only nil
+                              :same nil}
+                             {:left-only
+                              {:numeric_scale 0
+                               :column_default nil
+                               :character_octet_length nil
+                               :datetime_precision nil
+                               :nullable? "NO"
+                               :character_maximum_length nil
+                               :numeric_precision 64
+                               :numeric_precision_radix 2
+                               :data_type "bigint"
+                               :column_name "fact_value_id"
+                               :table_name "facts"}
+                              :right-only nil
+                              :same nil}
+                             {:left-only
+                              {:numeric_scale 0
+                               :column_default "nextval('fact_values_id_seq'::regclass)"
+                               :character_octet_length nil
+                               :datetime_precision nil
+                               :nullable? "NO"
+                               :character_maximum_length nil
+                               :numeric_precision 64
+                               :numeric_precision_radix 2
+                               :data_type "bigint"
+                               :column_name "id"
+                               :table_name "fact_values"}
+                              :right-only nil
+                              :same nil}
+                             ;; new columns
+                             {:left-only nil
+                              :right-only
+                              {:numeric_scale nil
+                               :column_default nil
+                               :character_octet_length nil
+                               :datetime_precision nil
+                               :nullable? "YES"
+                               :character_maximum_length nil
+                               :numeric_precision nil
+                               :numeric_precision_radix nil
+                               :data_type "bytea"
+                               :column_name "large_value_hash"
+                               :table_name "facts"}
+                              :same nil}
+                             ;; shrank value_type_id from bigint to integer
+                             {:left-only {:data_type "bigint" :numeric_precision 64}
+                              :right-only {:data_type "integer" :numeric_precision 32}
```

Comment:
Good point(s), though originally, I wasn't thinking about the id here as a pk, I was thinking of it (via domain specific knowledge) as a type tag that we know has specific, unchangable, integer values (https://github.com/puppetlabs/puppetdb/blob/57196c992dc68563532d16ae7c82f1081077b071/src/puppetlabs/puppetdb/scf/migrate.clj#L247) that we've been hard-coding on the clojure side (https://github.com/puppetlabs/puppetdb/blob/57196c992dc68563532d16ae7c82f1081077b071/src/puppetlabs/puppetdb/facts.clj#L146), and figured we could just use the smaller postgresql "best balance" type here -- but happy to leave that off if we'd rather, and wasn't thinking at the time about the mismatch created with the other table.